src/pkg/scan/postprocessors/report_converters.go

```diff
@@ -0,0 +1,201 @@
+	FromRelationalSchema(ctx context.Context, reportUUID string, reportSummary string) (string, error)
+}
+
+// nativeToRelationalSchemaConverter is responsible for converting the JSON scan report from the Harbor 1.0 format to
+// the generic vulnerability format which follows a normalized storage schema.
+type nativeToRelationalSchemaConverter struct {
+	dao scan.VulnerabilityRecordDao
+}
+
+// NewNativeToRelationalSchemaConverter returns a new instance of a V1 report to V2 report converter
+func NewNativeToRelationalSchemaConverter() NativeScanReportConverter {
+	return &nativeToRelationalSchemaConverter{dao: scan.NewVulnerabilityRecordDao()}
+}
+
+// ToRelationalSchema converts the vulnerability report data present as JSON  to the new relational VulnerabilityRecord instance
+func (c *nativeToRelationalSchemaConverter) ToRelationalSchema(ctx context.Context, reportUUID string, registrationUUID string, digest string, reportData string) (string, string, error) {
+
+	if len(reportData) == 0 {
+		log.Infof("There is no vulnerability report to toSchema for report UUID : %s", reportUUID)
+		return reportUUID, "", nil
+	}
+
+	// parse the raw report with the V1 schema of the report to the normalized structures
+	rawReport := new(vuln.Report)
+	if err := json.Unmarshal([]byte(reportData), &rawReport); err != nil {
+		return "", "", errors.Wrap(err, fmt.Sprintf("Error when toSchema V1 report to V2"))
+	}
+
+	if err := c.toSchema(ctx, reportUUID, registrationUUID, digest, reportData); err != nil {
+		return "", "", errors.Wrap(err, fmt.Sprintf("Error when converting vulnerability report"))
+	}
+	rawReport.Vulnerabilities = nil
+	data, err := json.Marshal(rawReport)
+	if err != nil {
+		return "", "", errors.Wrap(err, fmt.Sprintf("Error when persisting raw report summary"))
+	}
+	return reportUUID, string(data), nil
+
+}
+
+// FromRelationalSchema converts the generic vulnerability record stored in relational form to the
+// native JSON blob.
+func (c *nativeToRelationalSchemaConverter) FromRelationalSchema(ctx context.Context, reportUUID string, reportSummary string) (string, error) {
+	vulns, err := c.dao.GetForReport(ctx, reportUUID)
+	if err != nil {
+		return "", errors.Wrap(err, fmt.Sprintf("Error when toSchema generic vulnerability records for %s", reportUUID))
+	}
+	rp, err := c.fromSchema(ctx, reportUUID, reportSummary, vulns)
+	if err != nil {
+		return "", err
+	}
+	return rp, nil
+}
+
+func (c *nativeToRelationalSchemaConverter) toSchema(ctx context.Context, reportUUID string, registrationUUID string, digest string, rawReportData string) error {
+
+	var vulnReport vuln.Report
+	err := json.Unmarshal([]byte(rawReportData), &vulnReport)
+	if err != nil {
+		return err
+	}
+	for _, v := range vulnReport.Vulnerabilities {
+		vulnV2 := new(scan.VulnerabilityRecord)
+		vulnV2.CVEID = v.ID
+		vulnV2.Description = v.Description
+		vulnV2.Package = v.Package
+		vulnV2.PackageVersion = v.Version
+		vulnV2.Digest = digest
```

Comment:
The digest is always the digest of the artifact, so the column is redundancy in the model.

Reply:
Done. Removed digest and report_uuid columns